synapse/event_auth.py

```diff
@@ -646,27 +647,29 @@ def auth_types_for_event(event):
     actually auth the event.
     """
     if event.type == EventTypes.Create:
-        return []
+        return set()
 
-    auth_types = [
-        (EventTypes.PowerLevels, ""),
-        (EventTypes.Member, event.sender),
-        (EventTypes.Create, ""),
-    ]
+    auth_types = set(
```

Comment:
might be a shade more efficient to use a set literal rather than construct the set from a tuple?

```
auth_types = {
            (EventTypes.PowerLevels, ""),
            (EventTypes.Member, event.sender),
            (EventTypes.Create, ""),
}

Reply:
I keep forgetting about set tuples